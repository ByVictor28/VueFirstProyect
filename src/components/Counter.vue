<template>
  <h2>{{customTitle}}</h2>
  <p >{{counter}} <sup>2</sup> = {{counter*counter}}</p>
  <p data-testid="counter">{{counter}}</p>
  <hr>
  
  <div>
      <button @click="increase">+1</button>
      <button @click="decrease">-1</button> 
  </div>
</template>

<script>
export default {
    props: {
        title:String,
        start:{
            type:Number,
            required:true,
            default:1,
            validator(value){
                return value > 100
            }
        },
    },
    
    name:"Counter",
    data(){
        return {
            counter:this.start
        }
    },
    methods:{
        getSquereValue(){
            return this.counter * this.counter
        },
        increase(){
            this.counter = this.counter+1
        },
        decrease(){
            this.counter = this.counter-1
        }
    },
    computed:{
        squareCounter(){
            return this.counter * this.counter
        },
        customTitle(){
            return this.title || "Counter"
        }
    }
}
</script>

<style>
    button{
        background: #64bb87;
        margin: 0 5px;
        border-radius: 5px;
        color: white;
        padding: 5px 16px;
    }
    button:hover{
        background: #5aa67b;
        cursor: pointer;
    }
</style>