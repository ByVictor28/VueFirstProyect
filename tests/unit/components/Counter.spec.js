import { shallowMount, mount } from "@vue/test-utils"

import Counter from "@/components/Counter"
describe("Counter component",()=> {
  let wrapper
  let pCounter
  beforeEach(()=>{
    wrapper = shallowMount(Counter)
    pCounter = wrapper.find('[data-testid="counter"]')
  })

  // test("Must match with the snapshot", ()=> {
  //   expect(wrapper.html()).toMatchSnapshot()
  // })

  test("h2 must have default value",()=>{


    const h2 = wrapper.find("h2")

    expect(h2.text()).toBe("Counter")
  })

  test("p value must be 100",()=>{

    expect(pCounter.text()).toBe("1")

    // expect(h2.text()).toBe("Counter")
  })

  test("increase or decrease counter by 1",async ()=>{

    const buttonIncreace =  wrapper.findAll("button")[0]
    await buttonIncreace.trigger("click")
    expect(pCounter.text()).toBe("2")

    const buttonDecreace =  wrapper.findAll("button")[1]
    await buttonDecreace.trigger("click")
    await buttonDecreace.trigger("click")
    expect(pCounter.text()).toBe("0"); 
  })

  test("Must asign the default value",()=>{
    const {start} =  wrapper.props()

    expect(pCounter.text()).toBe(start.toString())
  })

  test("Must show prop title",()=>{
    const customTitle = "Hola mundo"
    const wrapper = shallowMount(Counter,{props:{title:customTitle}})

    expect( wrapper.find("h2").text() ).toBe(customTitle)    
  })
})