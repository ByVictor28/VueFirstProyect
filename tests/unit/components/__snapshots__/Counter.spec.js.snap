// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component Must match with the snapshot 1`] = `
<h2>Counter</h2>
<p>1 <sup>2</sup> = 1</p>
<hr>
<div><button>+1</button><button>-1</button></div>
`;
